### PR TITLE
feat(ctb): break out FSD steps

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Breaks out the FreshSystemDictator execution steps into a separate deployment stage and adds new post-deploy checks. Also fixes a bug in the L1CrossDomainMessenger initializer that only impacted fresh deployments.

### CODE REVIEW

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
with a bindings rebuild this should be good to merge

File: packages/contracts-bedrock/contracts/deployment/BaseSystemDictator.sol [hunk 1]
```diff
@@ -79,14 +79,14 @@ contract BaseSystemDictator is Ownable {
     /**
      * @notice Current step;
      */
-    uint256 public currentStep = 1;
+    uint8 public currentStep = 1;
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/deployment/BaseSystemDictator.sol:82]: What does the `uint8` get us?

**Review by maurelian — APPROVED** [review]
one question, but lgtm

File: packages/contracts-bedrock/contracts/deployment/BaseSystemDictator.sol (hunk 1 above)
Comment (smartcontracts) [reply packages/contracts-bedrock/contracts/deployment/BaseSystemDictator.sol:82]: Dumb reason, means the result from `ethers` isn't a BigNumber so I don't need to do `.toNumber()`. And we'll never have 255 steps here (dear lord I hope we never have 255 steps) so not a problem.

**Pull request merged**